projects/media-viewer/src/lib/store/effects/annotations.effect.ts

```diff
@@ -0,0 +1,57 @@
+import { Injectable } from '@angular/core';
+import { Actions, Effect, ofType } from '@ngrx/effects';
+import {catchError, exhaustMap, map, switchMap} from 'rxjs/operators';
+import { of } from 'rxjs';
+import {AnnotationApiService} from '../../annotations/annotation-api.service';
+import * as annotationsActions from '../actions';
+
+@Injectable()
+export class AnnotationEffects {
+  constructor(
+    private actions$: Actions,
+    private annotationApiService: AnnotationApiService,
+  ) { }
+
+  @Effect()
+  loadAnnotation$ = this.actions$.pipe(
+    ofType(annotationsActions.LOAD_ANNOTATION_SET),
+    map((action: annotationsActions.LoadAnnotationSet) => action.payload),
+    switchMap((url) => {
+      return this.annotationApiService.getAnnotationSet(url).pipe(
+        map(annotations => {
+          return new annotationsActions.LoadAnnotationSetSucess(annotations.body);
+        }),
+        catchError(error => {
+          return of(new annotationsActions.LoadAnnotationSetFail(error));
+        }));
+    }));
+
+  @Effect()
+  postAnnotation$ = this.actions$.pipe(
+    ofType(annotationsActions.SAVE_ANNOTATION),
+    map((action: annotationsActions.SaveAnnotation) => action.payload),
+    exhaustMap((annotation) => {
+      return this.annotationApiService.postAnnotation(annotation).pipe(
+        map(annotations => {
+          return new annotationsActions.SaveAnnotationSuccess(annotations);
+        }),
+        catchError(error => {
+          return of(new annotationsActions.LoadAnnotationSetFail(error));
+        }));
+    }));
+
+  @Effect()
+  deleteAnnotation$ = this.actions$.pipe(
+    ofType(annotationsActions.DELETE_ANNOTATION),
+    map((action: annotationsActions.DeleteAnnotation) => action.payload),
+    exhaustMap((annotation) => {
+      return this.annotationApiService.deleteAnnotation(annotation).pipe(
+        map(result => {
+          return new annotationsActions.DeleteAnnotationSucess(annotation);
+        }),
+        catchError(error => {
+          return of(new annotationsActions.DeleteAnnotationFail(error));
+        }));
+    }));
+}
```

Comment:
how come there is no call to create the annotationSet if it doesn't exist. we could have two separate actions, Load and Create to differentiate between when we're loading the document, and when we're trying to create annotations

Reply:
annotation set is now managed page page entities... so if there is no annotation page still creates them based on how many pages are displayed from pdf... how BE will managed that I am not sure